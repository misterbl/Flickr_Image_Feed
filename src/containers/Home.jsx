import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { getFlickrFeedItems, fetchingData } from "../selectors/apiSelectors";
import * as apiThunk from "../actions/thunks/apiThunk";
import FlickrCard from "../components/FlickrCard";

export class Home extends React.Component {
  componentDidMount() {
    const { flickrFeedItems, getFlickrFeed } = this.props;
    if (!flickrFeedItems) {
      getFlickrFeed();
    }
  }
  render() {
    return (
      <div className="flex flex-wrap ml6">
        {this.props.flickrFeedItems &&
          this.props.flickrFeedItems.map(item => (
            <FlickrCard
              image={item.media.m}
              title={item.title}
              author={item.author}
              author_id={item.author_id}
              link={item.link}
              description={item.description}
              tags={item.tags}
              key={item.title + item.published + item.media.m}
            />
          ))}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  flickrFeedItems: getFlickrFeedItems(state),
  fetchingData: fetchingData(state)
});

export const mapDispatchToProps = {
  getFlickrFeed: apiThunk.getFlickrFeed
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Home)
);
