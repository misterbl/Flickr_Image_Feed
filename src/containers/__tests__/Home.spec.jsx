import * as React from "react";
import { shallow } from "enzyme";
import { Home } from "../Home";
import generateAppState from "../../testMocks/appState.mock";

describe("Home", () => {
  const state = generateAppState();
  const props = {
    flickrFeedItems: state.api.feed.flickrFeed.items,
    getFlickrFeed: jest.fn(),
    fetchingData: false
  };

  const wrapper = shallow(<Home {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe("conponentDidMount", () => {
    it("doesn't call getFlickrFeed if flickrFeedItems is not null", () => {
      wrapper.instance();
      expect(props.getFlickrFeed).not.toHaveBeenCalled();
    });
    it("calls getFlickrFeed if flickrFeedItems null", () => {
      const newProps = { ...props, flickrFeedItems: null };
      const newWrapper = shallow(<Home {...newProps} />);
      newWrapper.instance();
      expect(newProps.getFlickrFeed).toHaveBeenCalled();
    });
    it("renders the FlickrCard if flickrFeedItems is not null", () => {
      const flickrCard = wrapper.find("FlickrCard");
      expect(flickrCard.length).toEqual(1);
    });
    it("doesn't render the FlickrCard but a h1 tag if ForumData is null", () => {
      const newProps = { ...props, flickrFeedItems: null };
      const newWrapper = shallow(<Home {...newProps} />);
      const flickrCard = newWrapper.find("FlickrCard");
      expect(flickrCard.length).toEqual(0);
    });
  });
});
