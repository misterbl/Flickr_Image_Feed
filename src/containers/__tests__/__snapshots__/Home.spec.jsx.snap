// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    fetchingData={false}
    flickrFeedItems={
      Array [
        Object {
          "media": Object {
            "author": "authorTest",
            "author_id": "123",
            "description": "descriptionTest",
            "link": "linkTest",
            "m": "mediaTest",
            "tags": "tagsTest",
            "title": "titleTest",
          },
        },
      ]
    }
    getFlickrFeed={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FlickrCard
          image="mediaTest"
        />,
      ],
      "className": "flex flex-wrap ml6",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "NaNmediaTest",
        "nodeType": "class",
        "props": Object {
          "author": undefined,
          "author_id": undefined,
          "description": undefined,
          "image": "mediaTest",
          "link": undefined,
          "tags": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FlickrCard
            image="mediaTest"
          />,
        ],
        "className": "flex flex-wrap ml6",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "NaNmediaTest",
          "nodeType": "class",
          "props": Object {
            "author": undefined,
            "author_id": undefined,
            "description": undefined,
            "image": "mediaTest",
            "link": undefined,
            "tags": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
